File: 2078.maxDistance.go
package main

func maxDistance(colors []int) int {
	size := len(colors)

	mostLeft := colors[0]
	mostRight := colors[size-1]

	best := 0

	for L := 0; L < size; L++ {
		if colors[L] != mostRight {
			best = size - 1 - L
			break
		}
	}

	for R := size - 1; R >= 0; R-- {
		if colors[R] != mostLeft {
			best = max(best, R)
			break
		}
	}

	return best
}

File: 2078.maxDistance.c
#include "shdstd.h"

#define MAX(x, y) (((x) > (y)) ? (x) : (y))
int maxDistance(int *colors, int size)
{
	int most_left = colors[0];
	int most_right = colors[size-1];

	int best = 0;
	for (int L = 0; L < size; ++L) {
		if (colors[L] != most_right) {
			best = size - 1 - L;
			break;
		}
	}

	for (int R = size-1; R >= 0; --R) {
		if (colors[R] != most_left) {
			best = MAX(best, R);
			break;
		}
	}

	return best;
}
